redfish-core/lib/account_service.hpp

```diff
@@ -1377,6 +1377,139 @@ inline void updateUserProperties(std::shared_ptr<bmcweb::AsyncResp> asyncResp,
+        [asyncResp](const boost::system::error_code ec,
+                    sdbusplus::message::message& m,
+                    const std::tuple<std::vector<uint8_t>, bool, std::string>&
+                        messageFDbus) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "DBUS response error: " << ec;
+                if (strcmp(m.get_error()->name,
+                           "xyz.openbmc_project.Certs.Error."
+                           "InvalidCertificate") == 0)
+                {
+                    redfish::messages::invalidUpload(
+                        asyncResp->res,
+                        "/redfish/v1/AccountService/Accounts/service",
+                        "Invalid Certificate");
+                }
+                else
+                {
+                    messages::internalError(asyncResp->res);
+                }
+                return;
+            }
+            getAcfProperties(asyncResp, messageFDbus);
+        },
+        "xyz.openbmc_project.Certs.ACF."
+        "Manager",
+        "/xyz/openbmc_project/certs/ACF", "xyz.openbmc_project.Certs.ACF",
+        "InstallACF", decodedAcf);
+}
+
+inline void triggerUnauthenticatedACFUpload(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    std::optional<nlohmann::json> oem)
+{
+    std::optional<nlohmann::json> ibm;
+    if (!redfish::json_util::readJson(*oem, asyncResp->res, "IBM", ibm))
+    {
+        BMCWEB_LOG_ERROR << "Illegal Property ";
+        messages::propertyMissing(asyncResp->res, "IBM");
+        return;
+    }
+
+    if (ibm)
+    {
+        std::optional<nlohmann::json> acf;
+        if (!redfish::json_util::readJson(*ibm, asyncResp->res, "ACF", acf))
+        {
+            BMCWEB_LOG_ERROR << "Illegal Property ";
+            messages::propertyMissing(asyncResp->res, "ACF");
+            return;
+        }
+
+        if (acf)
+        {
+            std::vector<uint8_t> decodedAcf;
+            std::optional<std::string> acfFile;
+            if (acf.value().contains("ACFFile") &&
+                acf.value()["ACFFile"] == nullptr)
+            {
+                acfFile = "";
+            }
+            else
+            {
+                if (!redfish::json_util::readJson(*acf, asyncResp->res,
+                                                  "ACFFile", acfFile))
+                {
+                    BMCWEB_LOG_ERROR << "Illegal Property ";
+                    messages::propertyMissing(asyncResp->res, "ACFFile");
+                    return;
+                }
+
+                std::string sDecodedAcf;
+                if (!crow::utility::base64Decode(*acfFile, sDecodedAcf))
+                {
+                    BMCWEB_LOG_ERROR << "base64 decode failure ";
+                    messages::internalError(asyncResp->res);
+                    return;
+                }
+                try
+                {
+                    std::copy(sDecodedAcf.begin(), sDecodedAcf.end(),
```

Comment:
Would like to avoid something that throws here but I think both of my suggestions still do. 


Why a std::copy here instead of just an assign? 
Like 
https://github.com/openbmc/bmcweb/blob/26f6976f3905d7b20690bbf794fba746c1929ed3/redfish-core/lib/event_service.hpp#L406 
or just the vector constructor 

std::vector<uint8_t> vec(str.begin(), str.end());

Reply:
Just re-used the already implemented and working code so did not change that.
https://github.com/ibm-openbmc/bmcweb/pull/243/files#diff-a6441dc86bbef5cbe2192c66717e8b938f9ee3a02670ff54c702338095a0d660L2103

Let me know if you still want me to change that.

Reply:
Ack.